### PR TITLE
Connect Video Chat Experiment

### PR DESCRIPTION
**Pull request by benhalpern**
## What type of PR is this? (check all applicable)

- [ ] Refactor
- [x] Feature
- [ ] Bug Fix
- [ ] Optimization
- [ ] Documentation Update

## Description

This is a half-built experiment on a new paradigm for doing video chat in `/connect`.

We have a prior experiment live but it never really took the next steps because it's somewhat of complicated mess. This would allow us to remove all that old code and actually build something more maintainable.

The idea is that the "call" happens in the sidecar, and you join the call by accessing the page propped up in the sidecar.

To create a new call, someone creates a message which opens a sidecar link. Then the call happens in the sidecar.

The actual code for the video area is the minimum amount for it to _maybe_ work. But I wanted to shoot it into prod to mess around there.

This is literally 25 minutes of work but I think could be nice to continue the theme of letting`/connect` worry about basic stuff and extra functionality should be driven through sidecar links. The `VideoChat` code is just mostly copy and pasted from the other stuff, but I think if this works we can tear out the rest and tell the `Chat` code it doesn't generally have to worry about special video logic anymore.

Screen Shot 2020-04-06 at 10 26 21 AM

### CODE REVIEW

File: app/javascript/packs/videoChat.jsx [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        token,
+        {
+          name: `private-video-channel-${activeChannelId}`,
+          audio: true,
+          type: 'peer-to-peer',
+          video: { width: 640 },
+        },
+      ).then(
+        function onConnectSuccess(room) {
+          console.log(room)
```
Comment (mstruve) [inline app/javascript/packs/videoChat.jsx:28]: Do we need all the console log statements?

File: app/javascript/packs/videoChat.jsx [hunk 2]
```diff
@@ -0,0 +1,119 @@
...
+                'The new dominant speaker in the Room is:',
+                dominantSpeaker,
+              );
+            });
+          });
+        },
+        function onConnectFailure(error) {
+          document.getElementById('videoremotescreen').innerHTML = '';
+          // eslint-disable-next-line no-console
+          console.error(`Unable to connect to Room: ${error.message}`);
```
Comment (mstruve) [inline app/javascript/packs/videoChat.jsx:69]: Not sure if this gets sent to Honeybadger or not, might be better to send this explicitly to Honeybadger so we know we can see it.

File: app/views/video_chats/show.html.erb [hunk 3]
```diff
@@ -0,0 +1,13 @@
+<div class="video-chat-wrapper" id="chat" data-token="<%= @token %>" data-channel="<%= params[:id] %>">
+
+</div>
+<h1>This is an experiment</h1>
+<h3>It's only a partially implemented view.</h3>
+<h4>But this could be a more straightforward way to do video chat within /connect.</h4>
+<%# TODO Make paramsp[:id] map to a chat channel and ensure the users have permission %>
```
Comment (mstruve) [inline app/views/video_chats/show.html.erb:7]: Before we do this do we want to restrict what user's can use this feature? Like maybe only trusted users or something.

**Comment by benhalpern** [comment]
@mstruve Since we actually kind of have a version of all this public already, and this is even a bit more hidden, I'm gonna go ahead and merge this. But with all this in mind as obvious stuff that needs to be worked out before anyone is truly using it.

There's no path within the app to really get here by accident and anyone that tries to do this on purpose it's not a big deal if it's kind of half-working.

**Pull request merged**